python/ray/tune/trial_runner.py

```diff
@@ -414,6 +411,17 @@ def _process_events(self):
             if trial.is_restoring:
                 with warn_if_slow("process_trial_restore"):
                     self._process_trial_restore(trial)
+            elif trial.is_saving:
+                with warn_if_slow("process_trial_save") as profile:
+                    self._process_trial_save(trial)
+                if profile.too_slow and trial.sync_on_checkpoint:
+                    # TODO(ujvl): Suggest using DurableTrainable once
+                    #  API has converged.
+                    logger.warning(
+                        "Consider turning off forced head-worker trial "
+                        "checkpoint syncs by setting sync_on_checkpoint=False"
+                        ". Note that this might result in faulty trial "
+                        "restoration for some worker failure modes.")
```

Comment:
Might want to elaborate on this. I wouldn't turn the forced syncs off if I didn't know what exactly could fail.